File: Web/src/api/ctf.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFList(data) {
  return apiClient.post("/ctf/GetCTFList", data).then(handleResponse).catch(handleError)
}

function GetCTF(data) {
  return apiClient.post("/ctf/GetCTF", data).then(handleResponse).catch(handleError)
}

function CreateCTF(data) {
  return apiClient.post("/ctf/CreateCTF", data).then(handleResponse).catch(handleError)
}

function UpdateCTF(data) {
  return apiClient.post("/ctf/UpdateCTF", data).then(handleResponse).catch(handleError)
}

function DeleteCTF(data) {
  return apiClient.post("/ctf/DeleteCTF", data).then(handleResponse).catch(handleError)
}

const ctfApi = {
  GetCTFList,
  GetCTF,
  CreateCTF,
  UpdateCTF,
  DeleteCTF
}

export default ctfApi

File: Server/route/ctf.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCTFRoute(r *gin.Engine) {
	adminCTF := r.Group("/ctf")
	adminCTF.Use(middleware.AdminAuthMiddleware())
	{
		adminCTF.POST("/GetCTFList", api.GetCTFList)
		adminCTF.POST("/GetCTF", api.GetCTF)
		adminCTF.POST("/CreateCTF", api.CreateCTF)
		adminCTF.POST("/UpdateCTF", api.UpdateCTF)
		adminCTF.POST("/DeleteCTF", api.DeleteCTF)
	}
}
